src_code/BoundlessBrilliance/Controller/PresentationListCollectionViewController.swift

```diff
@@ -59,27 +60,44 @@ class PresentationListCollectionViewController: UICollectionViewController, UICo
             print (presentationDict)
             
             if (presentationDict != nil){
-                self.loadDataIntoArray(presentationDict: presentationDict)
+                self.loadDataIntoArray(presentationDict: presentationDict, ref: ref)
             }
         })
     }
     
-    func loadDataIntoArray(presentationDict: [String : Dictionary<String, Any>]) {
+    func loadDataIntoArray(presentationDict: [String : Dictionary<String, Any>], ref: DatabaseReference) {
         var presenterDict: Dictionary<String, String>!
         for presentation in presentationDict.values {
             // values["presenters"] as! Dictionary<String, String>)
             presenterDict = presentation["presenters"] as? Dictionary
             let parsedPresenterString = parsePresenterDictionary(presenterNames: Array(presenterDict.values))
             print(parsedPresenterString)
             
-            
-            
             let date_string : String = presentation["date"] as! String
             let formatted_date : String = parseDateTime (datetime : date_string).0
             let formatted_time: String = parseDateTime(datetime : date_string).1
-         
-            self.presentationItems.append(PresentationListItemModel(location: presentation["location"] as! String, names: parsedPresenterString, chapter: presenterChapter, time: formatted_time, date: formatted_date))
-            self.collectionView!.reloadData()
+            
+            // Obtain userID from first presenter
+            var presentationChapter: String!
```

Comment:
Code to get chapter should be in its own function like parseDateTime

Reply:
and inside getChapterName() or whatever, that should call another function that gets the userID